app/models/project.rb

```diff
@@ -623,7 +620,7 @@ def check_status(deprecated_or_removed = false)
         update_attribute(:status, nil)
         update_attribute(:deprecation_reason, nil)
       end
-    elsif deprecated_or_removed # if package is already deprecated or removed and isn't anymore, remove status
+    elsif status == "Removed"
       update_attribute(:status, nil)
```

Comment:
Should this also update the deprecation_reason like on line 624?

Reply:
I believe that any package with a `deprecation_reason` set should fall into the "can_have_entire_package_deprecated" condition branch above. I think we'd only reach this scenario if a package was deprecated, was later incorrectly marked as "Removed," and was then found to be neither removed nor deprecated.

We could potentially clear out the deprecation reason here and also in the first couple "Removed" conditionals as well in case a previous deprecation is removed, but not sure if its worth the additional complexity.

Reply:
I think @havocp may have had thoughts about this, but also probably fine to do this in a separate effort.

Reply:
I think if I were writing this code from scratch I'd ignore the current status and deprecation reason and have code that just computed the new ones, and then I'd just set the new ones, and let ActiveRecord worry about whether they changed.

But what you have here I think is a fine minimal change to just get it to make sense.